### PR TITLE
LPS-133431 | master

### PR DESCRIPTION
**Pull request by ThalitaCeli**

### CODE REVIEW

**Comment by Tim-Cao** [comment]
Hi @ThalitaCeli, widget page template is for page. Widget Template is for widget. It was called ApplicationDisplayTemplate Could you move the case to ApplicationDisplayTemplateWithStaging.testcase?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/widgetpagetemplate/WidgetPageTemplates.testcase [hunk 1]
```diff
@@ -539,4 +539,73 @@ definition {
...
+			JSONGroup.addGroup(groupName = "Test Site Name");
+		}
+
+ 		task ("Add a Navigation Menu template") {
+			ApplicationDisplayTemplates.openWidgetTemplatesAdmin(siteURLKey = "test-site-name");
+
+			ApplicationDisplayTemplates.addADTWithFieldEntryCP(
+				portletName = "Navigation Menu",
+				fieldName = "Included Pages");
+		}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/widgetpagetemplate/WidgetPageTemplates.testcase]: Unneeded. Try to other OOTB display template

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/navigation/NavigationMenusWidget.path [hunk 2]
```diff
@@ -50,6 +50,11 @@
 	<td>//a[contains(@class,'dropdown-toggle')][contains(.,'${key_pageName}')]/parent::li[contains(@class,'open')]</td>
 	<td></td>
 </tr>
+<tr>
+	<td>NAVIGATION_MENU_TEMPLATE</td>
+	<td>//div[contains(@class,'portlet-body')][contains(.,'${key_fieldName}')]</td>
+	<td></td>
+</tr>
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/navigation/NavigationMenusWidget.path:57]: Unneeded

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/widgetpagetemplate/WidgetPageTemplates.testcase [hunk 3]
```diff
@@ -539,4 +539,73 @@ definition {
...
+				key_fieldName = "auto",
+				locator1 = "NavigationMenusWidget#NAVIGATION_MENU_TEMPLATE");
+		}	
+
+		task ("Publish the content page") {
+			PageEditor.clickPublish();
+		}
+
+		task ("Navigate to the site staging") {
+			Navigator.openSiteURL(siteName = "Test Site Name Staging");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/widgetpagetemplate/WidgetPageTemplates.testcase]: Use
```
			ContentPagesNavigator.openViewContentPage(
				pageName = "Test Page Name",
				siteName = "Test Site Name Staging");
```

**Comment by ThalitaCeli** [comment]
Hi @Tim-Cao,
I removed the unneeded path and the change I made in the other macro. I don't know if the file is correct that way. Could you tell me if I need to make any changes now?
Thank you!

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/applicationdisplaytemplate/ApplicationDisplayTemplatesWithStaging.testcase [hunk 4]
```diff
@@ -54,6 +54,59 @@ definition {
...
+				pageName = "Test Page Name",
+				siteName = "Test Site Name Staging");
+
+			PageEditor.addWidget(portletName = "Navigation Menu");
+		}
+
+		task ("Configure the Navigation Menu widget to use the Navigation Menu Template") {
+			NavigationMenusWidget.gotoNavigationMenuWidgetConfig();
+
+			NavigationMenusWidget.selectDisplayTemplate(templateName = "Navigation Menu Test ADT");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/applicationdisplaytemplate/ApplicationDisplayTemplatesWithStaging.testcase:89]: Use other OOTB display template, for example, List Menu, Tabs
image

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/applicationdisplaytemplate/ApplicationDisplayTemplatesWithStaging.testcase [hunk 5]
```diff
@@ -54,6 +54,59 @@ definition {
...
+	@priority = "4"
+	test CustomWidgetTemplateWithLocalLiveStaging {
+		task ("Add a widget templates in staging site") {
+			ApplicationDisplayTemplates.openWidgetTemplatesAdmin(siteURLKey = "test-site-name-staging");
+
+			var templateScript = '''<a>${content.getData()}</a>''';
+
+			ApplicationDisplayTemplates.addCP(
+				adtType = "Navigation Menu",
+				templateScript = "${templateScript}");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/applicationdisplaytemplate/ApplicationDisplayTemplatesWithStaging.testcase:67]: Unneeded steps

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/applicationdisplaytemplate/ApplicationDisplayTemplatesWithStaging.testcase [hunk 6]
```diff
@@ -54,6 +54,59 @@ definition {
...
+		}
+
+		task ("Configure the Navigation Menu widget to use the Navigation Menu Template") {
+			NavigationMenusWidget.gotoNavigationMenuWidgetConfig();
+
+			NavigationMenusWidget.selectDisplayTemplate(templateName = "Navigation Menu Test ADT");
+
+			Portlet.saveAndCloseConfig();
+		}
+		task ("Publish the content page") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/applicationdisplaytemplate/ApplicationDisplayTemplatesWithStaging.testcase:93]: Add step to wait for auto save

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/applicationdisplaytemplate/ApplicationDisplayTemplatesWithStaging.testcase [hunk 7]
```diff
@@ -54,6 +54,59 @@ definition {
...
+			Staging.openStagingAdmin(siteURLKey = "test-site-name-staging");
+
+			Staging.publishCustomPublication(checkContentNameList = "PortletDisplayTemplate");
+		}
+
+		task ("View the widget template in live site") {
+			ApplicationDisplayTemplates.openWidgetTemplatesAdmin(siteURLKey = "test-site-name");
+
+			ApplicationDisplayTemplates.viewTemplate(adtType = "Navigation Menu");
+		}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/applicationdisplaytemplate/ApplicationDisplayTemplatesWithStaging.testcase:107]: Should navigate to content page at view mode to check if the display template is applied to Navigation Menu widget in live site

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/applicationdisplaytemplate/ApplicationDisplayTemplatesWithStaging.testcase [hunk 8]
```diff
@@ -54,6 +54,59 @@ definition {
...
+			NavigationMenusWidget.gotoNavigationMenuWidgetConfig();
+
+			NavigationMenusWidget.selectDisplayTemplate(templateName = "Navigation Menu Test ADT");
+
+			Portlet.saveAndCloseConfig();
+		}
+		task ("Publish the content page") {
+			PageEditor.clickPublish();
+		}
+
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/applicationdisplaytemplate/ApplicationDisplayTemplatesWithStaging.testcase:96]: Add steps to navigate to content page at view mode to check if the display template is applied to Navigation Menu widget in staging site

**Review by Tim-Cao — COMMENTED** [review]
Hi @ThalitaCeli, Could you update case according to screenshot and comments? Thanks

**Review by Tim-Cao — COMMENTED** [review]
SF error

**Comment by ThalitaCeli** [comment]
> SF error

I will be shipping with SF. Thanks.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/applicationdisplaytemplate/ApplicationDisplayTemplatesWithStaging.testcase [hunk 9]
```diff
@@ -54,6 +54,64 @@ definition {
...
+			ContentPagesNavigator.openViewContentPage(
+				pageName = "Test Page Name",
+				siteName = "Test Site Name Staging");
+		}
+		task ("View the widget template applied to Navigation Menu widget in staging site") {
+			AssertTextEquals(
+				locator1 = "ControlMenu#HEADING",
+				value1 = "Test Page Name");
+		}
+		task ("Publish to Live") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/applicationdisplaytemplate/ApplicationDisplayTemplatesWithStaging.testcase:98]: Add step to switch to ready for publication

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/applicationdisplaytemplate/ApplicationDisplayTemplatesWithStaging.testcase [hunk 10]
```diff
@@ -54,6 +54,64 @@ definition {
 		}
 	}
 
+	@description = "This is for LPS-130051. Widget templates can be used to modify the look of a specific application."
+	@priority = "4"
+	test CustomWidgetTemplateWithLocalLiveStaging {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/applicationdisplaytemplate/ApplicationDisplayTemplatesWithStaging.testcase:59]: ApplyWidgetTemplateToNavigationMenuOnContentPage

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/applicationdisplaytemplate/ApplicationDisplayTemplatesWithStaging.testcase [hunk 11]
```diff
@@ -54,6 +54,64 @@ definition {
...
+		}
+		task ("Configure the Navigation Menu widget to use the List Menu Template") {
+			NavigationMenusWidget.gotoNavigationMenuWidgetConfig();
+
+			NavigationMenusWidget.selectDisplayTemplate(templateName = "List Menu");
+
+			Portlet.saveAndCloseConfig();
+		}
+		task ("Wait for auto save and Publish the content page") {
+			PageEditor.waitForAutoSave();
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/applicationdisplaytemplate/ApplicationDisplayTemplatesWithStaging.testcase:84]: Invalid step

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/applicationdisplaytemplate/ApplicationDisplayTemplatesWithStaging.testcase [hunk 12]
```diff
@@ -54,6 +54,64 @@ definition {
...
+		}
+		task ("Navigate to the content page") {
+			ContentPagesNavigator.openViewContentPage(
+				pageName = "Test Page Name",
+				siteName = "Test Site Name Staging");
+		}
+		task ("View the widget template applied to Navigation Menu widget in staging site") {
+			AssertTextEquals(
+				locator1 = "ControlMenu#HEADING",
+				value1 = "Test Page Name");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/applicationdisplaytemplate/ApplicationDisplayTemplatesWithStaging.testcase:96]:
```suggestion
			AssertVisible(
				key_itemName = "Test Page Name",
				locator1 = "NavigationMenusWidget#LIST_MENU_ANY_ITEM");
```

**Comment by Tim-Cao** [comment]
Hi @ThalitaCeli, see changes https://github.com/liferay-echo/liferay-portal/compare/f64a68d6e267122f66a1c6755de1ed654e8af075...effad5c03601256a882ac1532cbd12727babcdd3

**Comment by Tim-Cao** [comment]
ci:close

**Comment by ThalitaCeli** [comment]
> Hi @ThalitaCeli, see changes [liferay-echo/liferay-portal@f64a68d...a81a686](https://github.com/liferay-echo/liferay-portal/compare/f64a68d6e267122f66a1c6755de1ed654e8af075...a81a686c400d738e3140d3bdca2fa2987fd4661e)

Okay. Thank you!

**Pull request closed without merging**